tools/windows-node-installer/pkg/cloudprovider/azure/vm.go

```diff
@@ -482,12 +494,109 @@ func (az *AzureProvider) constructStorageProfile(imageId string) (storageProfile
 }
 
 // getWorkerBackendPoolID gets the backend pool id of the worker node loadbalancer
-func (az *AzureProvider) getWorkerBackendPoolID() *string {
-	// We assume that BackendPoolID for worker vm follows the following format
-	// /subscriptions/$SUBSCRIPTID/resourceGroups/$INFRAID-rg/providers/Microsoft.Network/loadBalancers/$INFRAID/backendAddressPools/$INFRAID
-	var wbpi = fmt.Sprintf("/subscriptions/%s/resourceGroups/%[2]s-rg/providers/"+
-		"Microsoft.Network/loadBalancers/%[2]s/backendAddressPools/%[2]s", az.subscriptionID, az.infraID)
-	return &wbpi
+func (az *AzureProvider) getWorkerBackendAddressPools(ctx context.Context) ([]network.BackendAddressPool, error) {
+	// Create a filter to get a resource of type virtualMachine which contains 'worker' in its name
+	filter := "resourceType eq 'Microsoft.Compute/virtualMachines' and substringof('worker', name)"
+	// number of resources to retrieve
+	// https://learnk8s.io/kubernetes-node-size
+	// Kubernetes can technically accommodate 5000 nodes, considering that for optimal performance organizations limit at 500 worker nodes
+	top := int32(500)
+	// Use getResource() function to get all the workerVMs by filtering VMs which have 'worker' in their name
+	vmResources, err := az.getResource(ctx, filter, top)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get worker node: %v", err)
+	}
+	if vmResources == nil {
+		return nil, fmt.Errorf("failed to get worker node VM resource")
+	}
+
+	// create a map to keep track of all the loadbalancers associated with different worker nodes.
+	lbSet := make(map[string]struct{})
+	// filler value for map
+	var exists = struct{}{}
```

Comment:
Why do you need a map when an array would suffice?

Reply:
I'm using a `map` to avoid duplicates. 
because golang doesn't have a `set` data structure. i'm implementing a `map` as a `set`
consider a scenario, where we have 10VMs. 
according to the code, it'll go through all the 10VMs one by one and will append their backend pool address to the `Map`. [L568](https://github.com/openshift/windows-machine-config-bootstrapper/blob/f6013f4116d675ef5b228b9b014a2f06cc5a9d03/tools/windows-node-installer/pkg/cloudprovider/azure/vm.go#L568)

Now, if we have backendpool `lb1` on VMs 1,2,4,5 
and backend piil `lb2` on VMs 3,6,7. 
in that case `map` helps us in avoid appending duplicates (using backend pool id as key). 
quick lookup of existing keys (used in test code). 
later i'm using the number of keys to determine if we have multiple worker load balancers attached [L571](https://github.com/openshift/windows-machine-config-bootstrapper/blob/f6013f4116d675ef5b228b9b014a2f06cc5a9d03/tools/windows-node-installer/pkg/cloudprovider/azure/vm.go#L571)

Reply:
If that is the logic you are following then you should stop the minute you find the 2nd worker behind a different LB. So you can use a single variable to track that.

Reply:
Accepted. I will add an `if` condition to stop when we get more than one lb. 
but this type of implementation makes it future proof, i.e if and when we decide to implement multiple LBs on windows VM. we'll just need to remove/add few `if` conditions and add a loop. Will also add comments in the code regarding this.